backend/dataall/modules/metadata_forms/db/metadata_form_models.py

```diff
@@ -0,0 +1,149 @@
+from sqlalchemy import (
+    Column,
+    String,
+    Integer,
+    ForeignKey,
+    Boolean,
+    ForeignKeyConstraint,
+    PrimaryKeyConstraint,
+)
+from sqlalchemy.dialects.postgresql import ARRAY
+from sqlalchemy.orm import relationship, validates
+
+from dataall.base.db import Base, utils
+
+from dataall.modules.metadata_forms.db.enums import MetadataFormFieldType
+
+
+class MetadataForm(Base):
+    __tablename__ = 'metadata_form'
+    uri = Column(String, primary_key=True, default=utils.uuid('metadata_form'))
+    name = Column(String, nullable=False)
+    description = Column(String, nullable=False)
+    SamlGroupName = Column(String, nullable=True)
+    visibility = Column(String, nullable=False)  # enum MetadataFormVisibility
+    homeEntity = Column(String, nullable=True)
+
+
+class MetadataFormEnforcementRule(Base):
+    __tablename__ = 'metadata_form_enforcement_rule'
+    uri = Column(String, primary_key=True, default=utils.uuid('rule'))
+    metadataForm = Column(String, ForeignKey('metadata_form.uri'))
+    level = Column(String, nullable=False)  # enum MetadataFormEnforcementScope
+    entityTypes = Column(ARRAY(String), nullable=False)  # enum MetadataFormEntityTypes
+    severity = Column(String, nullable=False)  # enum MetadataFormEnforcementSeverity
+
+
+class MetadataFormField(Base):
+    __tablename__ = 'metadata_form_field'
+    metadataForm = Column(String, ForeignKey('metadata_form.uri'))
+    uri = Column(String, primary_key=True, default=utils.uuid('field'))
+    name = Column(String, nullable=False)
+    type = Column(String, nullable=False)  # enum MetadataFormFieldType
+    required = Column(Boolean, nullable=False)
+    glossaryTerm = Column(String, ForeignKey('term_link.linkUri'), nullable=True)
```

Comment:
should it be TermLink or GlossaryNode?

Reply:
We want to link the field with root glossary nodes. AFAIK, this root glossary nodes are connected with terms. Am i right?

Reply:
TermLink is used to associate a particular Term with a target resource Uri 

GlossaryNode is used to represent a Glossary Resource (more specifically Glossary Node can be of Type 'G' (Glossary Root), 'C' (Category), or 'T' (Term))

I think what you may want here is Glossary Nodes - and really only Glossary nodes of node type either C or G... and then the possibleValues would auto-populate as the Terms in the category or in the entire glossary

Reply:
[98beaeb](https://github.com/data-dot-all/dataall/pull/1422/commits/98beaeb072716fad57a1f7c93b31930e2ca09981)

Restrictions on NodeType will be implemented programmatically